tests/CLITestCase.php

```diff
@@ -2,79 +2,74 @@
 
 namespace tests\ClearcodeHQ\CommandBusLauncherBundle;
 
-use Symfony\Bundle\FrameworkBundle\Client;
 use Symfony\Bundle\FrameworkBundle\Console\Application;
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
 use Symfony\Component\Console\Command\Command;
-use Symfony\Component\Console\Input\StringInput;
-use Symfony\Component\Console\Output\StreamOutput;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\Console\Tester\CommandTester;
 
 abstract class CLITestCase extends WebTestCase
 {
-    /** @var Client */
-    private $client;
-
-    /** @var ContainerInterface */
-    private $container;
+    /** @var int */
+    protected $statusCode;
+    /** @var string */
+    protected $display;
 
     /** {@inheritdoc} */
     public static function getKernelClass()
     {
-        include_once __DIR__ . '/App/TestKernel.php';
+        include_once __DIR__.'/App/TestKernel.php';
+
         return 'tests\ClearcodeHQ\CommandBusLauncherBundle\App\TestKernel';
     }
 
-    /**
-     * @return Client
-     */
-    protected function client()
+    /** {@inheritdoc} */
+    protected function setUp()
     {
-        return $this->client;
+        $this->prepareKernel();
     }
 
     /**
-     * @return ContainerInterface
+     * @param Command $command
+     * @param array   $parameters
      */
-    protected function container()
+    protected function executeCommand(Command $command, $parameters = [])
     {
-        return $this->container;
-    }
+        $application = new Application(static::$kernel);
+        $application->add($command);
 
-    /**
-     * {@inheritdoc}
-     */
-    protected function setUp()
-    {
-        parent::setUp();
-        $this->client    = $this->createClient();
-        $this->container = $this->client->getContainer();
+        $tester = new CommandTester($command);
+        $tester->execute($parameters);
+
+        $this->display = $tester->getDisplay();
+        $this->statusCode = $tester->getStatusCode();
     }
 
-    /**
-     * {@inheritdoc}
-     */
+    /** {@inheritdoc} */
     protected function tearDown()
     {
-        $this->container = null;
-        $this->client    = null;
+        $this->display = null;
+        $this->statusCode = null;
 
         parent::tearDown();
     }
 
-    /**
-     * @param Command $command
-     * @param array $parameters
-     */
-    protected function executeCommand(Command $command, $parameters = [])
+    protected function assertThatStatusCodeEquals($statusCode)
     {
-        $application = new Application($this->container()->get('kernel'));
-        $application->add($command);
+        $this->assertEquals($statusCode, $this->statusCode);
+    }
 
-        $tester = new CommandTester($command);
+    protected function assertThatOutputWasDisplayed()
```

Comment:
Maybe we should add more extended version of this assert like `assertThatOutputContains($string)`.